sjb/config/test_cases/ci-kubernetes-machine-api-operator.yml

```diff
@@ -0,0 +1,232 @@
+---
+parent: 'common/test_cases/minimal.yml'
+extensions:
+  actions:
+    - type: "host_script"
+      title: "Upload the default AWS credentiasl"
+      script: |-
+         ssh -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config openshiftdevel "mkdir ~/.aws"
+         scp -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config ~/.aws/credentials openshiftdevel:~/.aws
+    - type: "script"
+      title: "Install minikube"
+      script: |-
+        # Install minikube
+        curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.28.0/minikube-linux-amd64
+        chmod +x minikube
+        sudo mv minikube /usr/bin/
+        # Install kubectl
+        curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl
+        chmod +x kubectl
+        sudo mv kubectl /usr/bin/
+    - type: "script"
+      title: "Deploy kubernetes"
+      script: |-
+        sudo setenforce 0
+        # https://github.com/kubernetes/minikube/blob/master/docs/configuring_kubernetes.md
+        sudo minikube start --vm-driver=none --extra-config=kubelet.cgroup-driver=systemd
+    - type: "script"
+      title: "Build the machine api operator"
+      script: |-
+        # build the mao image
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/machine-api-operator
+        go build -v -o bin/machine-api-operator github.com/openshift/machine-api-operator/cmd
+        go get -u github.com/openshift/imagebuilder/cmd/imagebuilder
+        sudo mv /data/bin/imagebuilder /usr/bin
+        sudo imagebuilder -t "docker.io/machine-api-operator:$(git describe --always --abbrev=7)" .
+    - type: "script"
+      title: "Deploy the machine api operator"
+      script: |-
```

Comment:
@ingvagabund Cannot we move this to `machine-api-operator` repository. This script looks generic enough that it could be used also locally.

Reply:
I would prefer to put all of it under `test/e2e/deploy_mao.go` or similar file, specify the manifests via native kubernetes types and query the apiserver the usual way. All implemented in Go. So, this is a temporary solution so we can at least test the mao deploys correctly. We can extend and improve the testing pieces later.

Reply:
can we just store the yaml manifests you are creating inline in to the main repo instead, otherwise as soon as we change the operator config the job will fail. Once we start to rewrite the long term implementation we can think about deploy_mao.go